### DISCUSSION TITLE
How to achieve loop solving？

### DISCUSSION THREAD

**Discussion by yongxin3344520** [Z3Prover/z3#6843] [Q&A]
How to achieve loop solving like this?

Ex:

// int x;
// int y;

for (int i=0;  i<x; ++i){
    y++ ;
}

assert(y == 100 )

**Comment by NikolajBjorner** [comment]
the loop exit condition is y = 0.

**Reply by yongxin3344520** [reply]
Yes, you are right !
But I found that I made a mistake in the Line1 .  Now I have corrected it. Let's start over again .
